tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,757 @@
+
+        # Wait for all uploads to complete: this should take roughly
+        # segment_max_upload_interval_sec plus manifest_max_upload_interval_sec
+        quiesce_uploads(self.redpanda, [self.topic],
+                        timeout_sec=self.params.manifest_upload_interval +
+                        self.params.segment_upload_interval +
+                        self.params.grace_upload_wait_interval)
+
+        # Check manifest upload metrics:
+        #  - we should not have uploaded the manifest more times
+        #    then there were manifest upload intervals in the runtime.
+        self.logger.info(
+            f"Upload counts: {stats['manifest_uploads']} manifests, "
+            f"{stats['segment_uploads']} segments")
+        chk.check_non_zero_value(stats['manifest_uploads'],
+                                 value_label="manifest_uploads")
+        chk.check_non_zero_value(stats['segment_uploads'],
+                                 value_label="segment_uploads")
+
+        # +3 because:
+        # - 1 empty upload at start
+        # - 1 extra upload from runtime % upload interval
+        # - 1 extra upload after the final interval_sec driven uploads
+        expect_manifest_uploads = (
+            (int(produce_duration) // self.params.manifest_upload_interval) +
+            3)
+        self.logger.info(f"Manifests uploads: {stats['manifest_uploads']}, "
+                         f"Expected not less than: {expect_manifest_uploads}")
+        chk.check_expected_manifest_uploads(stats["manifest_uploads"],
+                                            produce_duration)
+
+        # Do all required checks with Redpanda active
+        chk.conduct_checks()
+
+        # Run rp_storage_tool
+        self.redpanda.stop_and_scrub_object_storage(run_timeout=1800)
+        # Validate checks and generate AssertionError if needed
+        if chk.assert_results():
+            self.logger.info(chk.get_summary_as_text())
+
+    @cluster(num_nodes=6, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    # @parametrize(rolling_restarts=True)
+    @parametrize(rolling_restarts=False)
+    def long_retention_with_spillover_test(self, rolling_restarts):
+        self._iterative_retention_flow(rolling_restarts=rolling_restarts)
+
+    # Based on
+    # https://github.com/redpanda-data/redpanda/pull/10498/commits
+    @cluster(num_nodes=6,
+             log_allow_list=RESTART_LOG_ALLOW_LIST +
+             FAILURE_INJECTION_LOG_ALLOW_LIST)
+    @parametrize(batchtypes=[batchtype_archival_meta])
+    @parametrize(
+        batchtypes=[batchtype_raft_data, batchtype_raft_configuration])
+    def long_retention_with_spillover_and_failure_injection_test(
+            self, batchtypes):
+        batch_types_list = [getattr(BatchType, s) for s in batchtypes]
+        self._iterative_retention_flow(failure_injection_enabled=True,
+                                       batchtypes=batch_types_list)
+
+    def _iterative_retention_flow(self,
```

Comment:
nit: for other readers of this code, for such a long test, it'd be really helpful to describe at a high level what the test does, and what it checks